iidm/iidm-reducer/src/main/java/com/powsybl/iidm/reducer/SubNetworkPredicate.java

```diff
@@ -0,0 +1,115 @@
+        }
+        VoltageLevel vl = network.getVoltageLevel(voltageLevelId);
+        if (vl == null) {
+            throw new PowsyblException("Voltage level '" + voltageLevelId + "' not found");
+        }
+        Set<String> voltageLevelIds = new LinkedHashSet<>();
+        traverse(vl, 0, maxDepth, voltageLevelIds);
+        delegate = new IdentifierNetworkPredicate(voltageLevelIds);
+    }
+
+    private static void traverse(VoltageLevel vl, int depth, int maxDepth, Set<String> voltageLevelIds) {
+        if (voltageLevelIds.contains(vl.getId()) || depth > maxDepth) {
+            return;
+        }
+
+        voltageLevelIds.add(vl.getId());
+
+        vl.visitEquipments(new DefaultTopologyVisitor() {
+            public void visitBranch(Branch branch, Branch.Side side) {
+                switch (side) {
+                    case ONE:
+                        traverse(branch.getTerminal2().getVoltageLevel(), depth + 1, maxDepth, voltageLevelIds);
+                        break;
+                    case TWO:
+                        traverse(branch.getTerminal1().getVoltageLevel(), depth + 1, maxDepth, voltageLevelIds);
+                        break;
+                    default:
+                        throw new IllegalStateException("Unknown side: " + side);
+                }
+            }
+
+            @Override
+            public void visitLine(Line line, Line.Side side) {
+                visitBranch(line, side);
+            }
+
+            @Override
+            public void visitTwoWindingsTransformer(TwoWindingsTransformer transformer, TwoWindingsTransformer.Side side) {
+                visitBranch(transformer, side);
+            }
+
+            @Override
+            public void visitThreeWindingsTransformer(ThreeWindingsTransformer transformer, ThreeWindingsTransformer.Side side) {
+                switch (side) {
+                    case ONE:
+                        traverse(transformer.getLeg2().getTerminal().getVoltageLevel(), depth + 1, maxDepth, voltageLevelIds);
+                        traverse(transformer.getLeg3().getTerminal().getVoltageLevel(), depth + 1, maxDepth, voltageLevelIds);
+                        break;
+                    case TWO:
+                        traverse(transformer.getLeg1().getTerminal().getVoltageLevel(), depth + 1, maxDepth, voltageLevelIds);
+                        traverse(transformer.getLeg3().getTerminal().getVoltageLevel(), depth + 1, maxDepth, voltageLevelIds);
+                        break;
+                    case THREE:
+                        traverse(transformer.getLeg1().getTerminal().getVoltageLevel(), depth + 1, maxDepth, voltageLevelIds);
+                        traverse(transformer.getLeg2().getTerminal().getVoltageLevel(), depth + 1, maxDepth, voltageLevelIds);
+                        break;
+                    default:
+                        throw new IllegalStateException("Unknown side: " + side);
+                }
+            }
```

Comment:
You should visit HVDC lines also?

Reply:
Good question. In my own use case, I didn't want to traverse HVDC lines. I can imagine in some other use case, it would make sens.
What should I do?
  - let as it is and if somebody wants one day, we will consider to add it.
  - optionally add it right now

Reply:
It makes sense to do it now. Do we consider it's an option enabled by a boolean or do we consider it always traverse the HVDC lines